sweep/walletsweep.go

```diff
@@ -66,17 +66,21 @@ func DetermineFeePerKw(feeEstimator chainfee.Estimator,
 
 		return feePerKw, nil
 
-	// If a manual sat/byte fee rate is set, then we'll use that directly.
-	// We'll need to convert it to sat/kw as this is what we use
-	// internally.
+	// If a manual sat/byte fee rate is set, then we'll attempt to use that
+	// directly. We'll need to convert it to sat/kw as this is what we use
+	// internally. If the specified fee rate is lower than the relay fee
+	// rate, we'll use the relay fee rate instead. "Relay fee rate" may mean
+	// slightly different things depending on the backend. For bitcoind, it
+	// effectively means max(relay fee, min mempool fee).
 	case feePref.FeeRate != 0:
 		feePerKW := feePref.FeeRate
-		if feePerKW < chainfee.FeePerKwFloor {
+		minFeePerKW := feeEstimator.RelayFeePerKW()
+		if feePerKW < minFeePerKW {
 			log.Infof("Manual fee rate input of %d sat/kw is "+
 				"too low, using %d sat/kw instead", feePerKW,
-				chainfee.FeePerKwFloor)
+				minFeePerKW)
 
-			feePerKW = chainfee.FeePerKwFloor
+			feePerKW = minFeePerKW
```

Comment:
Maybe we should disallow creating transactions instead of defaulting to the min relay fee? Maybe I'd choose to not open a channel if the min relay fee is too high, but I don't know until I open the channel. Not sure tho, cc @guggero

Reply:
Hmm, good question... Maybe failing with an appropriate error message is actually the better option UX wise.

Reply:
I think most RPCs disallow setting 0 or a fraction of sat/vByte. So the existing behavior would only come into place in RPCs where we have sat/kWU (which isn't the case here). But if we want to be 100% exact, we would split the check into two: if the fee rate is below the absolute floor value, bump it. If it's still lower than the min relay/mempool fee, return an error (probably mentioning the original, user-set value, not the bumped one).

Reply:
> I think most RPCs disallow setting 0 or a fraction of sat/vByte. So the existing behavior would only come into place in RPCs where we have sat/kWU (which isn't the case here).

This makes sense.

> split the check into two: if the fee rate is below the absolute floor value, bump it. If it's still lower than the min relay/mempool fee, return an error (probably mentioning the original, user-set value, not the bumped one).

I would imagine in most instances the relay/mempool fee would be >= the absolute floor value. Is that accurate? If so, it doesn't seem very useful to introduce additional complexity to handle the case where the inequality is reversed (and a bumped fee would actually go through). What do you think?

Reply:
> I would imagine in most instances the relay/mempool fee would be >= the absolute floor value.

Yes, that's correct. I was just thinking about RPCs that use sat/kWU, if they also go through `DetermineFeePerKw` then users could actually enter a value lower than the absolute floor value. In that cases it would make sense to just bump that to the floor. But returning an error in both cases probably makes sense to keep the code simple.
We should probably be pretty specific in the error message, mentioning something along the lines of "fee of %d sat/vByte is too low to be accepted into the mempool or relayed to the network".